nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/EncodeContent.java

```diff
@@ -129,31 +172,35 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         }
     }
 
-    private static StreamCallback getStreamCallback(final boolean encode, final String encoding) {
-        if (encode) {
-            if (encoding.equalsIgnoreCase(BASE64_ENCODING)) {
-                return new EncodeBase64();
-            } else if (encoding.equalsIgnoreCase(BASE32_ENCODING)) {
-                return new EncodeBase32();
-            } else {
-                return new EncodeHex();
-            }
-        } else {
-            if (encoding.equalsIgnoreCase(BASE64_ENCODING)) {
-                return new DecodeBase64();
-            } else if (encoding.equalsIgnoreCase(BASE32_ENCODING)) {
-                return new DecodeBase32();
-            } else {
-                return new DecodeHex();
-            }
+    private static StreamCallback getStreamCallback(final boolean encode, final EncodingType encoding,
+        final int lineLength, final String lineSeparator) {
+        switch(encoding) {
+            case BASE64:
+                return encode ? new EncodeBase64(lineLength, lineSeparator) : new DecodeBase64();
+            case BASE32:
+                return encode ? new EncodeBase32(lineLength, lineSeparator) : new DecodeBase32();
+            default:
+                return encode ? new EncodeHex() : new DecodeHex();
         }
     }
 
     private static class EncodeBase64 implements StreamCallback {
 
+        private int lineLength;
+        private String lineSeparator;
```

Comment:
Intellij points out these should be final
```suggestion
       private final int lineLength;
       private final String lineSeparator;
```